src/olympia/git/management/commands/git_extraction.py

```diff
@@ -67,6 +68,15 @@ def extract_addon(self, entry, batch_size=BATCH_SIZE):
         addon = entry.addon
         log.info('Starting git extraction of add-on "{}".'.format(addon.pk))
 
+        # See: https://github.com/mozilla/addons-server/issues/14289
+        if addon.type != amo.ADDON_EXTENSION:
```

Comment:
Can we avoid them entering the queue at all? Even avoid entering the `extract_version_to_git` task ?

Which reminds me: let's not do this now, but ultimately, when we remove the code to extract without using the cron, we'll probably want to replace calls to `extract_version_to_git.delay` at upload/signing time entirely and replace them with just a function call doing the `GitExtractionEntry.objects.create(addon=version.addon)` bit, to avoid unnecessarily triggering tasks for just a single database insert.

Reply:
> Can we avoid them entering the queue at all? Even avoid entering the `extract_version_to_git` task ?

Yes, we could do this once we do what you said next: once we remove `extract_version_to_git.delay`, we'll have to create a `enqueue_git_extraction` or something and that's where we'll prevent non-webextensions to be added to the queue.

With this patch, we'll run `extract_version_to_git.delay` but non-webextensions won't be added to the queue anymore.

Reply:
Yeah that works for me.

Reply:
I filed https://github.com/mozilla/addons-server/issues/14311